src/components/Common/UI/Switch/Switch.test.tsx

```diff
@@ -0,0 +1,123 @@
+import { describe, expect, it, vi } from 'vitest'
+import { render, screen, within } from '@testing-library/react'
+import { userEvent } from '@testing-library/user-event'
+import type { RefObject } from 'react'
+import { createRef } from 'react'
+import { Switch } from './Switch'
+
+describe('Switch', () => {
+  const defaultProps = {
+    label: 'Test Switch',
+    name: 'test-switch',
+  }
+
+  it('renders correctly with label', () => {
+    render(<Switch {...defaultProps} />)
+    expect(screen.getByText('Test Switch')).toBeInTheDocument()
+  })
+
+  it('associates label with switch via htmlFor', () => {
+    render(<Switch {...defaultProps} />)
+    const labelElement = screen.getByText('Test Switch')
+    const switchElement = screen.getByRole('switch')
+    expect(labelElement).toHaveAttribute('for', switchElement.id)
+  })
+
+  it('associates error message with switch via aria-describedby', () => {
+    const errorMessage = 'This field is required'
+    render(<Switch {...defaultProps} errorMessage={errorMessage} />)
+
+    const switchElement = screen.getByRole('switch')
+    const errorMessageId = switchElement.getAttribute('aria-describedby')
+    expect(screen.getByText(errorMessage)).toHaveAttribute('id', errorMessageId)
+  })
+
+  it('associates description with switch via aria-describedby', () => {
+    const description = 'Helpful description'
+    render(<Switch {...defaultProps} description={description} />)
+
+    const switchElement = screen.getByRole('switch')
+    const descriptionId = switchElement.getAttribute('aria-describedby')
+    expect(screen.getByText(description)).toHaveAttribute('id', descriptionId)
+  })
+
+  it('calls onChange handler when clicked', async () => {
+    const user = userEvent.setup()
+    const onChange = vi.fn()
+
+    render(<Switch {...defaultProps} onChange={onChange} />)
+    const switchElement = screen.getByRole('switch')
+
+    await user.click(switchElement)
+
+    expect(onChange).toHaveBeenCalledTimes(1)
+    expect(onChange).toHaveBeenCalledWith(true)
+  })
+
+  it('shows as selected when checked is true', () => {
+    render(<Switch {...defaultProps} checked={true} />)
+
+    const switchWrapper = screen.getByRole('switch').closest('label')
+    expect(switchWrapper).toHaveAttribute('data-selected', 'true')
+  })
+
+  it('shows as not selected when checked is false', () => {
+    render(<Switch {...defaultProps} checked={false} />)
+
+    const switchWrapper = screen.getByRole('switch').closest('label')
+    expect(switchWrapper).not.toHaveAttribute('data-selected', 'true')
+  })
```

Comment:
Nit: on these ones i might recommend verifying that the underlying `input` element has the checked attribute set similar to tests for checkbox. That would make these tests more agnostic to react aria's implementation